include/linker/ramfunc.ld

```diff
@@ -0,0 +1,17 @@
+SECTION_DATA_PROLOGUE(.ramfunc,,)
+{
+#ifdef CONFIG_CPU_HAS_MPU
+	. = ALIGN( 1 << LOG2CEIL(__ramfunc_ram_end - __ramfunc_ram_start));
+#endif
+	__ramfunc_ram_start = .;
+	KERNEL_INPUT_SECTION(.ramfunc)
+	KERNEL_INPUT_SECTION(".ramfunc.*")
+#ifdef CONFIG_CPU_HAS_MPU
+	. = ALIGN(32);
+	. = ALIGN( 1 << LOG2CEIL(__ramfunc_ram_end - __ramfunc_ram_start));
+#endif
+	__ramfunc_ram_end = .;
```

Comment:
In order to make sure that the end of the MPU region does not overlap with the adjacent one after that, you need to align the end address as well with the minimum MPU alignment.